lib/elixir/src/elixir_map.erl

```diff
@@ -86,16 +86,46 @@ validate_match_key(Meta, List, E) when is_list(List) ->
 validate_match_key(_, _, _) ->
   ok.
 
+validate_not_repeated(Meta, Key, Used, E) ->
```

Comment:
In my opinion, we don't need to worry about this level of complexity. We can just check for exact matches, which will catch literals and keys in the same line, and that's good enough.

Reply:
It makes sense, I fixed it adding the whole key to the `Used`

Reply:
I just realized this approach may make this match though: `%{next() => 1, next() => 2}` where `next()` is based on a side-effect. So I do think we need a quick literal check though. Something like:

```erlang
is_literal({_, _, _)) -> false;
is_literal({Left, Right}) -> is_literal(Left) andalso is_literal(Right);
is_literal([_ | _] = List) -> lists:all(fun is_literal/1, List);
is_literal(_) -> true.
```